sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOperation.cs

```diff
@@ -60,17 +63,21 @@ public class RecognizeCustomFormsOperation : Operation<IReadOnlyList<RecognizedF
         /// </summary>
         /// <param name="operations"></param>
         /// <param name="diagnostics"></param>
-        /// <param name="modelId"></param>
         /// <param name="operationLocation"></param>
-        internal RecognizeCustomFormsOperation(ServiceClient operations, ClientDiagnostics diagnostics, string modelId, string operationLocation)
+        internal RecognizeCustomFormsOperation(ServiceClient operations, ClientDiagnostics diagnostics, string operationLocation)
         {
             _serviceClient = operations;
             _diagnostics = diagnostics;
-            _modelId = modelId;
 
             // TODO: Add validation here
             // https://github.com/Azure/azure-sdk-for-net/issues/10385
-            Id = operationLocation.Split('/').Last();
+
+            string[] substrs = operationLocation.Split('/');
+
+            _resultId = substrs[substrs.Length - 1];
+            _modelId = substrs[substrs.Length - 3];
+
+            Id = string.Join("/", substrs, substrs.Length - 3, 3);
```

Comment:
This feels a little "magic numbery" to me ... can we do this for P1 & file an issue to use a Regex for it for Preview 3?

Reply:
#11505